### DISCUSSION TITLE
meson.options' choice only support string and combo?

### DISCUSSION THREAD

**Discussion by Freed-Wu** [mesonbuild/meson#13140] [Q&A]
Can we

```meson
option(
  'bit_depth',
  type: 'combo',
  choices: [8, 10],
  description: 'bit depth, can be 8 or 10',
)
```

Or

```meson
option(
  'bit_depth',
  type: 'integer',
  choices: [8, 10],
  description: 'bit depth, can be 8 or 10',
)
```

**Comment by dcbaker** [comment]
Combos should be able to hold integers as well as strings. If not then you can always do the conversion from string to integer in the `meson.build` file (`get_option('bit_depth').to_int()`)

The option types them selves are [documented here](https://mesonbuild.com/Build-options.html#build-options), and there are integers, arrays, combos, features, booleans, and strings.
